### PR TITLE
[4.3] Accessibility Features Plugin: choice of two different icon images (Google Font)

### PR DESCRIPTION
**Pull request by sandewt**
Pull Request for Issue #38832 .
(second attempt)

### Summary of Changes

- Choice of two icon images (Google Font)
- Choice of a rectangular (default) or a circular shape image

### Testing Instructions

- Enable the 'System - Additional Accessibility Features' plugin
- Select the "Use Google Material Font"
- Test the different options, see the attached images
- Code review

### Actual result BEFORE applying this Pull Request

No option to choose from two different icons and a shape (Google Font)

### Expected result AFTER applying this Pull Request

pr_accessibility-2

pr_accessibility-3
pr_accessibility-4png

### Some notes
- It has not researched of the "package": "accessibility", "version": "3.0.17" is still correct.

See links: 
- [https://ranbuch.github.io/accessibility/](https://ranbuch.github.io/accessibility/)
- [https://ranbuch.github.io/accessibility/site/](https://ranbuch.github.io/accessibility/site/)

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### Open questions
**- It is still Not possible to change the icon in the 'IconEmojis' mode**
**- It will require some documentation udates**

[EDIT]

### CODE REVIEW

File: plugins/system/accessibility/accessibility.php [hunk 1]
```diff
@@ -99,6 +99,8 @@ public function onBeforeCompileHead()
                         ],
                     ],
                     'useEmojis' => $this->params->get('useEmojis') != 'false' ? true : false,
+                    'img' => $this->params->get('image'),
+                    'circular' => $this->params->get('image_circular') != 'false' ? true : false,
```
Comment (bembelimen) [inline plugins/system/accessibility/accessibility.php]:
```suggestion
                    'circular' => $this->params->get('image_circular') != 'false',
```

**Comment by sandewt** [comment]
Maybe it makes more sense to change the Radio form field type to a List form field type?

@bembelimen @laoneo What do you think of that?

àccessibility_3

**Comment by laoneo** [comment]
Good suggestion.

**Comment by sandewt** [comment]
> Good suggestion.

Done

**Comment by sandewt** [comment]
Addition information: see the source code in browser (info for code review)

Use Emjois

```
"icon": {
            "position": {
                "left": {
                    "size": "0",
                    "units": "px"
                }
            },
            "useEmojis": true,
            "img": "",
            "circular": ""
        },
```		
		
Use Google Material Font
	
```
"icon": {
            "position": {
                "left": {
                    "size": "0",
                    "units": "px"
                }
            },
            "useEmojis": false,
            "img": "accessible",
            "circular": false
        },
```

**Comment by sandewt** [comment]
@drmenzelit  

Related to "Update accessibility plugin to version 4.5.3 #39161

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.0-dev. No new features will be merged into Joomla! 4.3 series. Joomla! 4.4 series is a bridge release to make migration from Joomla! 4 to 5 as smooth as possible.

**Comment by rachelwalraven** [comment]
I have tested this item :white_check_mark: successfully on 3b1b15459a3aad6ba5099833d15c6505f36f2e14<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39615">issues.joomla.org/tracker/joomla-cms/39615</a>.</sub>

**Comment by rachelwalraven** [comment]
I have tested this item :white_check_mark: successfully on 3b1b15459a3aad6ba5099833d15c6505f36f2e14<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39615">issues.joomla.org/tracker/joomla-cms/39615</a>.</sub>

**Comment by hvdmeer** [comment]
I have tested this item :red_circle: unsuccessfully on 3b1b15459a3aad6ba5099833d15c6505f36f2e14<br /><br />Tested this at the PBF in Nijmegen in the online J5 server  and got this error in Patch tester when applying the patch

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39615">issues.joomla.org/tracker/joomla-cms/39615</a>.</sub>

**Comment by brianteeman** [comment]
@hvdmeer that was because there are conflicting files as whon in your screenshot

**Comment by hvdmeer** [comment]
YEs @brianteeman , I talked with @sandewt about this PR as he was also at the PBF. Just added my failed test for future reference

**Comment by sandewt** [comment]
> Is this problem going to be fixed:

This PR is from last January, so it has been open for a while. If I have time, I will fix this code next week.

But there is still more work to be done, see PR #39161

Moreover, a newer version of the js is current. Maybe someone will want to fix that. My knowledge is insufficient to do that...

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by fgsw** [comment]
I have tested this item :red_circle: unsuccessfully on 3b1b15459a3aad6ba5099833d15c6505f36f2e14<br /><br />The file marked for modification does not exist: plugins/system/accessibility/accessibility.php<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39615">issues.joomla.org/tracker/joomla-cms/39615</a>.</sub>

**Comment by crimle** [comment]
I have tested this item :red_circle: unsuccessfully on 3b1b15459a3aad6ba5099833d15c6505f36f2e14<br /><br />Button [Apply patch] results in error message
The file marked for modification does not exist: plugins/system/accessibility/accessibility.php<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39615">issues.joomla.org/tracker/joomla-cms/39615</a>.</sub>

**Comment by exlemor** [comment]
@sandewt 

I have a question {perhaps a stupid one, I apologize if that's the case}, but while I love the idea of having choices for Google Fonts icons...  the only way for using Google Fonts to be GDPR compliant is for that font to be locally hosted... and unless I missed something during my check of the Joomla 5 installation, there is no local such font file [.ttf or .woff2]. So, sure it is great to have a choice of icons but not if that means not being GPDR compliant.

So, if we are going to do this PR, can we update the PR to include a download/installation of the Google Fonts file and/or include that Google Fonts file as part of the Joomla installation?

**Comment by brianteeman** [comment]
ignoring the misconception about google fonts not being gdpr compliant this PR is now redundant and should be closed as larger updates to this plugin are in progress which will allow a local font.

**Comment by Quy** [comment]
Closing in favor of #43143. Thank you.

**Comment by sandewt** [comment]
Thanks all

**Pull request closed without merging**